Lib/Inc/adc_data.h

```diff
@@ -0,0 +1,33 @@
+/**
+ * @file adc_data.h
+ */
+
+/*
+ *  Created on: Feb 14, 2022
+ *      Author: niuslar
+ */
+
+#ifndef ADC_DATA_H_
+#define ADC_DATA_H_
+
+#include "adc.h"
+#include "log.h"
+
+class CAdc
+{
+public:
+    CAdc(ADC_HandleTypeDef *p_hadc, uint8_t adc_channels);
+    ~CAdc();
+
+    // Public methods
+    // use [] operator to read a particular ADC Channel
+    uint16_t operator[](uint8_t adc_channel);
+    void triggerAdc();
+
+private:
+    ADC_HandleTypeDef *mp_hadc;
+    uint8_t m_adc_channels;
```

Comment:
this should be `const` to make it safer against someone modifying your code later.